npm-package/src/controllers/otelController.ts

```diff
@@ -1,104 +1,105 @@
-// let attributeArr = req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes[7]
-// let size = 0;
-// if (attributeArr?.key === 'http.response_content_length_uncompressed') {
-//     size = attributeArr.value.intValue;
-// }
-
-// console.log('size', size);
-
-//type
-// console.log('type', req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes[0]);
-// let typeArr = req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes[0];
-// // JSON.parse(typeArr);
-// let headers;
-// if (typeArr?.key === 'request-headers') {
-//     headers = JSON.parse(typeArr.value.stringValue);
-// }
-
-// console.log(headers?.accept);
-
-
-module.exports = otelController;
\ No newline at end of file
+function unixNanoToMS(unixNano: number): number {
+    const unixMS = Math.floor(unixNano / 1e6);
+    return unixMS;
+}
+
+
+export const otelController: OtelControllerType = {
+  parseAllRequest: (req: Request, res: Response, next: NextFunction) => {
+    let data: data = {}
+
+    if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId) {
+        data.traceId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId
+      }
+  
+      if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId) {
+        data.spanId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId
+      }
+  
+      if (req.body.resourceSpans[0].resource.attributes[4]?.value?.stringValue) {
+        data.applicationType = req.body.resourceSpans[0].resource.attributes[4].value.stringValue;
+      }
+  
+      if (req.body.resourceSpans[0].resource?.attributes[0]?.value?.stringValue) {
+        data.originatingService = req.body.resourceSpans[0].resource.attributes[0]?.value?.stringValue;
+      }
+
+      if (req.body.resourceSpans[0].scopeSpans[0]?.spans[0]?.name) {
+        data.method = req.body.resourceSpans[0].scopeSpans[0].spans[0].name;
+      }
+
+      const attributeArr = req.body.resourceSpans[0].scopeSpans[0].spans[0].attributes;
+      for (let i = 0; i < attributeArr.length; i++) {
+        if (attributeArr[i].key === 'http.status_code')
+          data.status = attributeArr[i].value.intValue;
+      }
+
+      for (let i = 0; i < attributeArr.length; i++) {
+        if (attributeArr[i].key === 'http.flavor'){
+          data.protocol = attributeArr[i].value.stringValue;
+          break;
+        }
+      }
+  
+    // if (req.body.resourceSpans[0].resource.attributes[4]?.value?.stringValue === 'next.js') {
+    //   data = otelController.parseNextJSRequest(req)
+    // }   
+    
+    if (req.body.resourceSpans[0].resource.attributes[4]?.value?.stringValue === 'node.js') {
+      data = otelController.parseNodeRequest(req, data);
+    }
+
+    data.startTime = unixNanoToMS(req.body.resourceSpans[0].scopeSpans[0].spans[0].startTimeUnixNano);
+    data.endTime = unixNanoToMS(req.body.resourceSpans[0].scopeSpans[0].spans[0].endTimeUnixNano);
+
+    console.log('final',data);
```

Comment:
I think we can remove extra console logs we made for our own development testing purposes if this isn't needed